pkg/provisioner/ironic/ironic.go

```diff
@@ -122,13 +122,13 @@ func LogStartup() {
 
 // A private function to construct an ironicProvisioner (rather than a
 // Provisioner interface) in a consistent way for tests.
-func newProvisioner(host *metal3v1alpha1.BareMetalHost, bmcCreds bmc.Credentials, publisher provisioner.EventPublisher) (*ironicProvisioner, error) {
-	clientIronic, err := clients.IronicClient(ironicEndpoint, ironicAuth)
+func newProvisionerWithSettings(host *metal3v1alpha1.BareMetalHost, bmcCreds bmc.Credentials, publisher provisioner.EventPublisher, ironicURL string, ironicAuthSettings clients.AuthConfig, inspectorURL string, inspectorAuthSettings clients.AuthConfig) (*ironicProvisioner, error) {
```

Comment:
That looks really nice. What about at this point replacing the other occurrences of `newProvisioner` in `ironic_test.go` with `newProvisionerWithSettings`? It will avoid setting explicitly the `IRONIC_ENDPOINT` and `IRONIC_INSPECTOR_ENDPOINT` environment variables to allow the other tests to pass.

Reply:
I updated the other calls. The variables still have to be set because the init function requires them. Refactoring that is going to be a bit more complicated so I would rather save that for a separate PR.

Reply:
Right, that's fine then.